providers/client_socket.js

```diff
@@ -122,6 +127,15 @@ ClientSocket.prototype.write = function(data) {
   this.outputStream.write(stringData, stringData.length);
 };
 
+ClientSocket.prototype.pause = function() {
+  this.paused = true;
+};
+
+ClientSocket.prototype.resume = function() {
+  this.paused = false;
+  this.rawInputStream.asyncWait(this.inputStreamCallback, 0, 0, mainThread);
```

Comment:
I don't know what to do with any of those flags.  I expect the defaults are fine.  If you want I can add a TODO to investigate...